spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/support/DefaultPublisherFactory.java

```diff
@@ -56,9 +59,17 @@
 
 	private final CredentialsProvider credentialsProvider;
 
-	public DefaultPublisherFactory(String projectId, ExecutorProvider executorProvider,
-			ChannelProvider channelProvider, CredentialsProvider credentialsProvider) {
-		this.projectId = projectId;
+	public DefaultPublisherFactory(GcpProjectIdProvider projectIdProvider,
+			ExecutorProvider executorProvider,
+			ChannelProvider channelProvider,
+			CredentialsProvider credentialsProvider) {
+		Assert.notNull(projectIdProvider, "The project ID provider can't be null.");
+		Assert.notNull(executorProvider, "The executor provider can't be null.");
+		Assert.notNull(channelProvider, "The channel provider can't be null.");
+		Assert.notNull(credentialsProvider, "The credentials provider can't be null.");
+
+		this.projectId = projectIdProvider.getProjectId()
+				.orElseThrow(MissingProjectIdException::new);
```

Comment:
??? Seems for me we agreed that we don't need `MissingProjectIdException` anymore to be consistent with the Spring assertion framework.